### ISSUE TITLE
Last modified date with label expressions

### ISSUE THREAD

**Issue by mjardemalm** [archimatetool/archi-modelrepository-plugin#175]
A nice additional feature for the already great function Label Expression would be to display the date of the last modification of a view or concept like `$view{modifieddate}`.
 
I have basic knowledge of how parts of Archi and coArchi works so i figure this would not be a straight forward implementation.
 
Is there a way to get Archi to hook up with events from a plugin? I guess this will require a more extensive implementation with at hook-editor and a set of pre-defined actions within Archi.
 
If a hook "Saved Object" are registered from coArchi and connected to a "Set Propery" action the existing label expressions will already cover the actual feature request.
 
Just a thought I’d like to share. The actual feature request would give me some great benefit. The thougths about a more generic solution just popped out of my head and could be totally ignored.

**Comment by Phillipus** [comment]
I'll move this to coArchi...

This isn't something we're going to implenment in coArchi. In fact, coArchi 2 won't be using lots of XML files and will commit the whole *.archimate file.

**Comment by mjardemalm** [comment]
Oh! That will be a game changer. Then git probably won't be useful to get track of changes anymore. If not diff could extract changed parts of the whole *.archimate file.

**Comment by Phillipus** [comment]
We will implement a logical diff, not a textual diff. Git is still useful for everything else.

**Comment by mjardemalm** [comment]
Interesting. Then maybe the logical diff could be used to track changed concepts and views. I'll guess time will tell what coArchi 2 will offer. =)

**Comment by mjardemalm** [comment]
> In this case what would be needed is a way for a plugin to contribute new expressions (but in this case performances would be an issue)

I fully agree to this. And in fact, I felt that I just had to implement a proof-of-concept for this approach. A format-patch is attached, code is based on commit 627203c. It's a new renderer that responds to `$prefex{commit:date|user}`.

I've been running it for almost a week on the same fairly large model without experiencing any latency issues.
 
It is based on the naming convention used for new objects where the object identifier and the type of object is used for identfy the XML-file that hosts the versioned object on disk. The renderer applies for concepts, relationships and diagrams. Uncommited objects will show the label expression itself.
 
Valid prefixes are model, view, source, target, parent and all connection prefixes. The model prefix will read the last log entry for whole repository where the others will read the last log entry for the specific object.

[commit-label-expression.patch.zip](https://github.com/archimatetool/archi-modelrepository-plugin/files/7469319/commit-label-expression.patch.zip)
